Add chat templates for command-a and aya-23-8B models

hyeobiiii opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

# Description

This pull request adds Hugging Face–compatible `chat_template` entries for the following models:

- `CohereLabs/c4ai-command-a-03-2025`
- `aya-23-8B`

## Motivation and Context

#### 1. c4ai-command-a-03-2025

- The original `tokenizer_config.json` for this model defines multiple templates (e.g., "default", "tool_use", "rag") as an array, which is not supported by Hugging Face’s `apply_chat_template()` method.
- The template also includes unsupported Jinja2 control structures such as `{% break %}`, causing runtime failures during prompt formatting.

#### 2. aya-23-8B

- The existing `cohere` template in Axolotl is designed for the `command-r` variant, which differs in prompt structure and token usage.
- To avoid ambiguity, a dedicated `aya` template is added to support `aya-23-8B` and future `aya`-based variants.


<!--- If it fixes an open issue, please link to the issue here. -->

## Related Issue

Fixes #2730

## How has this been tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

Instruct tuning was successfully conducted using the added `chat_template` for both `command-a` and `aya-23-8B` models. No formatting errors occurred, and the training proceeded as expected.

## Screenshots (if appropriate)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

## Social Handles (Optional)

<!-- Thanks for submitting a bugfix or enhancement. -->
<!-- We'd love to show our thanks to you on Twitter & Discord if you provide your handle -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced four new chat templates: "command_a", "command_a_tool_use", "command_a_rag", and "aya", offering advanced formatting for chat interactions, tool use, and document grounding.
  - Expanded chat template selection options with enhanced support for multi-step reasoning and a new assistant persona ("Aya").
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hyeobiiii committed:
Add chat templates for command-a and aya model

NanoCode012 commented:
Thanks for the PR and testing out the templates.

Just a quick look, there's also a few other changes noticed from the original (beside the for loop):
- `{{ tool_msg.content|tojson }}` -> `{{ tool_msg.content }}`
- `{{ tool['function']['parameters']|tojson }}` -> `{{ tool['function']['parameters']['properties']|tojson }}`

Could you please explain these change and any others that you may have made as well?

hyeobiiii committed:
Fix: isolate for-loop update and remove unintended changes

hyeobiiii commented:
@NanoCode012 

I realized that some unintended changes were mixed into the previous commit by mistake during the process. I've corrected it now — apologies for any confusion caused.
I’ve also updated the commit to only modify the for loop as intended.

NanoCode012 commented:
Thanks for the fix!
